### PR TITLE
Add deployment of BM-Inventory Monitoring stack

### PR DESCRIPTION
**Pull request by jparrill**
## Added Monitoring for BareMetal Inventory

This PR will include these list of features for development purposes (initially).

- [x] Added OLM Deployment in K8s cases.
- [x] Added OLM Prometheus in OCP and K8s cases.
- [x] Added Grafana in OCP and K8s cases.
- [x] Added Makefile Functions
- [x] Added Documentation

### CODE REVIEW

File: deploy/monitoring/prometheus/assisted-installer-ocp-prometheus-subscription-instance.yaml [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+    prometheus: assisted-installer-prometheus
+  namespace: assisted-installer
+spec:
+  replicas: 1
+  version: v2.8.0
+  serviceAccountName: prometheus-k8s
+  serviceMonitorSelector:
+    matchLabels:
+      app: assisted-installer-monitor
+  containers:
```
Comment (filanov) [inline deploy/monitoring/prometheus/assisted-installer-ocp-prometheus-subscription-instance.yaml:15]: need to add resources limitations

File: tools/deploy_prometheus.py [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+    src_file = os.path.join(os.getcwd(), "deploy/monitoring/prometheus/assisted-installer-ocp-prometheus-custom-ca.yaml")
+    dst_file = os.path.join(os.getcwd(), "build/assisted-installer-ocp-prometheus-custom-ca.yaml")
+    topic = 'OCP Custom CA'
+    with open(src_file, "r") as src:
+        with open(dst_file, "w+") as dst:
+            data = src.read()
+            data = data.replace("BASE64_CERT", ca_cert)
+            print("Deploying {}: {}".format(topic, dst_file))
+            dst.write(data)
+            dst.close
```
Comment (filanov) [inline tools/deploy_prometheus.py:59]: if you are using `with` you don't really need the close call.

File: tools/deploy_prometheus.py [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+
+def main():
+    if args.target != "oc-ingress":
+        # Deploy Operator Group
+        deployer('deploy/monitoring/prometheus/assisted-installer-operator-group.yaml', 'OperatorGroup')
+        # Deploy Subscription
+        deploy_prometheus_sub(olm_ns, cat_src)
+        # Deploy Prometheus Instance
+        deployer('deploy/monitoring/prometheus/assisted-installer-k8s-prometheus-subscription-instance.yaml', 'Prometheus Instance on K8s')
+        sleep(10)
```
Comment (filanov) [inline tools/deploy_prometheus.py:94]: can it be replaces with some validation?

Comment (jparrill) [reply tools/deploy_prometheus.py:109]: Yes, the validation is `utils.check_k8s_rollout('statefulset', 'prometheus-assisted-installer-prometheus')` the thing is, the prometheus operator, takes some time to process the CRD and with that sleep we are just giving them the time to the reconcile loop to reach the CRD parsing phase

Comment (filanov) [reply tools/deploy_prometheus.py:109]: Maybe something like https://pypi.org/project/waiting/?
it's not a blocker just trying to find a nice way to handle it.

Comment (jparrill) [reply tools/deploy_prometheus.py:109]: I've been reviewing the library and I cannot pass a function with arguments to wait, then I've implemented a more accurated function in utils.py.

**Comment by filanov** [comment]
> Should it be part of deploy-all?

It was a question, i'm not sure myself. 
@ronniel1 @avishayt what do you think? 
We can have it as part of deploy all or we can have it in a different target as the UI.

File: tools/utils.py [hunk 4]
```diff
@@ -36,7 +39,56 @@ def apply(file):
...
     cmd = '{kubecmd} get ingresscontrollers.operator.openshift.io -n openshift-ingress-operator -o custom-columns=:.status.domain'.format(kubecmd=KUBECTL_CMD)
     return check_output(cmd).split()[-1]
+
+
+def check_k8s_rollout(k8s_object, k8s_object_name, namespace="assisted-installer"):
+    cmd = '{} rollout status {}/{} --namespace {}'.format('kubectl', k8s_object, k8s_object_name, namespace)
+    return check_output(cmd)
+
+
+def wait_for(k8s_object, k8s_object_name, namespace="assisted-installer", limit=10, desired_status="successfully rolled out"):
```
Comment (filanov) [inline tools/utils.py:53]: maybe wait for rollout?

**Comment by jparrill** [comment]
For me it's fine, maybe we could maintain it in a separated flow. Up to you ;)

**Pull request merged**